index.d.ts

```diff
@@ -52,6 +52,6 @@ export declare function createEpicMiddleware<T extends Action, S, D = any>(rootE
 export declare function combineEpics<T extends Action, S, D = any>(...epics: Epic<T, S, D>[]): Epic<T, S, D>;
 export declare function combineEpics<E>(...epics: E[]): E;
 
-export declare function ofType<T extends Action>(...keys: T['type'][]): (source: Observable<T>) => Observable<T>;
+export declare function ofType<T extends Action, R extends T, K extends R['type'] = R['type']>(...key: K[]): (source: Observable<T>) => Observable<R>;
```

Comment:
I believe this would be a breaking change since it now requires two params instead of one. But if we provide a default param for the second one I think that would make it non-breaking and have the same behavior as before wouldn't it? `R extends T = T`

```ts
ofType<T extends Action, R extends T = T, K extends R['type'] = R['type']>
```

Reply:
yeah, good catch! my bad thx!